crypto/rand/randfile.c

```diff
@@ -280,8 +280,16 @@ const char *RAND_file_name(char *buf, size_t size)
     struct stat sb;
 #endif
 
+    /* Find the directory where the state file should go. */
+#ifdef OPENSSL_SYS_WINDOWS
+    if ((s = getenv("RANDFILE")) == NULL) {
+        s = getenv("TEMP");
+        if (s == NULL)
+            s = getenv("APPDATA");
```

Comment:
Better to remove the C:\ default than to set a bad location instead. Don't use DEFAULT_HOME, and return NULL if we don't have either RANDFILE or TEMP.